### PR TITLE
Restructure filters

### PR DESCRIPTION
**Pull request by WindQAQ**
As discussed in #178.

### CODE REVIEW

File: tensorflow_addons/image/filters_test.py [hunk 1]
```diff
@@ -18,74 +18,67 @@
...
         with self.assertRaisesRegexp(TypeError,
                                      'Filter shape must be a tuple'):
-            md.median_filter_2D(tf_img, 3)
-            md.median_filter_2D(tf_img, 3.5)
-            md.median_filter_2D(tf_img, 'dt')
-            md.median_filter_2D(tf_img, None)
+            median_filter2d(tf_img, 3)
+            median_filter2d(tf_img, 3.5)
+            median_filter2d(tf_img, 'dt')
+            median_filter2d(tf_img, None)
```
Comment (WindQAQ) [inline tensorflow_addons/image/filters_test.py]: Just found this is not robust. As long as one exception is raised within the context manager, the test will pass.

File: tensorflow_addons/image/filters_test.py [hunk 2]
```diff
@@ -18,74 +18,67 @@
...
         with self.assertRaisesRegexp(ValueError, msg):
-            md.median_filter_2D(tf_img, filter_shape)
+            median_filter2d(tf_img, filter_shape)
 
         with self.assertRaisesRegexp(TypeError,
                                      'Size of the filter must be Integers'):
-            md.median_filter_2D(tf_img, (3.5, 3))
-            md.median_filter_2D(tf_img, (None, 3))
+            median_filter2d(tf_img, (3.5, 3))
+            median_filter2d(tf_img, (None, 3))
```
Comment (WindQAQ) [inline tensorflow_addons/image/filters_test.py]: And this.

File: tensorflow_addons/image/filters_test.py [hunk 3]
```diff
@@ -18,74 +18,67 @@
...
         tf_img1 = tf.compat.v1.placeholder(tf.int32, shape=[3, None, 4])
         tf_img2 = tf.compat.v1.placeholder(tf.int32, shape=[None, 3, 4])
 
         with self.assertRaises(TypeError):
-            md.median_filter_2D(tf_img)
-            md.median_filter_2D(tf_img1)
-            md.median_filter_2D(tf_img2)
+            median_filter2d(tf_img)
+            median_filter2d(tf_img1)
+            median_filter2d(tf_img2)
```
Comment (WindQAQ) [inline tensorflow_addons/image/filters_test.py]: And this.

**Comment by seanpmorgan** [comment]
cc @Mainak431 for visibility

**Pull request merged**